mono/mini/driver.c

```diff
@@ -1601,6 +1601,13 @@ mono_jit_parse_options (int argc, char * argv[])
 			MonoDebugOptions *opt = mini_get_debug_options ();
 
 			opt->break_on_exc = TRUE;
+		} else if (strcmp (argv [i], "--test-tailcall-require") == 0) {
+			// This is an internal testing feature.
+			// Every tail. encountered is required to be optimized.
+			// It is asserted. There is no implied compatibility
+			// with this switch and it will be altered as desired
+			// to fit test needs. It is not in usage.
+			mini_get_debug_options ()->test_tailcall_require = TRUE;
```

Comment:
To set mini debug options, we use `MONO_DEBUG=...`. Also, adding a command line argument make it a breaking change to remove it, so it is less than ideal.

Reply:
It is not a breaking change as it is not in the usage. Removing environment variable support is breaking too. Oh, I guess you mean because the parser has to stay, and do nothing? That is ok.

Reply:
(and it only has to stay while you wait for existing internal users to drain, like until the test/makefile is edited away, which could/should be simultaneous

Reply:
There's no need for this. Use MONO_DEBUG

Reply:
This isn't for debugging actually. It is for testing.. which is funny really -- make check is a public thing. But we don't have good test infra here, so this is a wierd one instead. My other tests fit the usual models much better -- they run and exit() appropriately.